aea/configurations/data_types.py

```diff
@@ -451,13 +455,10 @@ def __repr__(self) -> str:
         return f"<{self}>"
 
     def __eq__(self, other: Any) -> bool:
-        """Compare with another object."""
-        return (
-            isinstance(other, PublicId)
-            and self.author == other.author
-            and self.name == other.name
-            and self.version == other.version
-        )
+        """Check equality."""
+        if not isinstance(other, self.__class__):
+            return NotImplemented  # Delegate comparison to the other instance's __eq__.
+        return all(getattr(self, s) == getattr(other, s) for s in self.__slots__[:-1])
```

Comment:
including the `_package_hash` will affect the functionality in the following 4 places, which is why I chose not to include it now. We may choose to open an issue on this if we want to change this later.
- `eject`
https://github.com/valory-xyz/open-aea/blob/77474ff2a02f50f218540418b0e760206c6a4360/aea/cli/eject.py#L204-L206
- `remove`
https://github.com/valory-xyz/open-aea/blob/77474ff2a02f50f218540418b0e760206c6a4360/aea/cli/remove.py#L193-L226
- `check_packages` 
https://github.com/valory-xyz/open-aea/blob/77474ff2a02f50f218540418b0e760206c6a4360/aea/cli/check_packages.py#L309-L310
- `registry/add`
https://github.com/valory-xyz/open-aea/blob/77474ff2a02f50f218540418b0e760206c6a4360/aea/cli/registry/add.py#L31-L72

Reply:
Please open an issue for this and add a comment in code.

Reply:
#449